server/hedley/modules/custom/hedley_admin/scripts/generate-demographics-report.php

```diff
@@ -0,0 +1,389 @@
+  ")->fetchAllAssoc('type');
+}
+
+/**
+ * Gathers group encounter patients by type.
+ *
+ * @return array
+ */
+function group_encounter_unique() {
+  return db_query("
+  SELECT
+  field_group_type_value as type, COUNT(*) as counter
+FROM
+(
+  SELECT
+      field_group_type_value,
+      p.field_person_target_id,
+      sess_rel.field_session_target_id
+    FROM
+      field_data_field_session sess_rel
+        LEFT JOIN field_data_field_clinic c ON sess_rel.field_session_target_id = c.entity_id
+        LEFT JOIN field_data_field_group_type gt ON field_clinic_target_id = gt.entity_id
+        LEFT JOIN field_data_field_person p ON p.entity_id = sess_rel.entity_id
+        LEFT JOIN person_classified class ON p.field_person_target_id = class.entity_id
+    WHERE
+        sess_rel.bundle IN ('attendance', 'birth_plan', 'breast_exam', 'child_fbf', 'contributing_factors',
+    'core_physical_exam',
+    'danger_signs', 'family_planning', 'follow_up', 'group_health_education',
+    'group_send_to_hc', 'height',
+    'lactation', 'last_menstrual_period', 'medical_history', 'medication',
+    'mother_fbf', 'muac', 'nutrition',
+    'nutrition_height', 'nutrition_muac', 'nutrition_nutrition', 'nutrition_photo',
+    'nutrition_weight',
+    'obstetric_history', 'obstetric_history_step2', 'obstetrical_exam', 'photo',
+    'pregnancy_testing',
+    'prenatal_family_planning', 'prenatal_health_education', 'prenatal_nutrition',
+    'prenatal_photo',
+    'resource', 'social_history', 'vitals', 'weight', 'acute_findings',
+    'acute_illness_danger_signs',
+    'acute_illness_follow_up', 'acute_illness_muac', 'acute_illness_nutrition',
+    'acute_illness_vitals',
+    'call_114', 'exposure', 'hc_contact', 'health_education', 'isolation',
+    'malaria_testing',
+    'medication_distribution', 'send_to_hc', 'symptoms_general', 'symptoms_gi',
+    'symptoms_respiratory',
+    'travel_history', 'treatment_history', 'treatment_ongoing',
+    'participant_consent', 'nutrition_caring',
+    'nutrition_contributing_factors', 'nutrition_feeding', 'nutrition_follow_up',
+    'nutrition_food_security',
+    'nutrition_health_education', 'nutrition_hygiene', 'nutrition_send_to_hc',
+    'appointment_confirmation',
+    'prenatal_follow_up', 'prenatal_send_to_hc', 'counseling_session')
+  AND field_group_type_value IS NOT NULL
+  AND class.entity_id IS NOT NULL
+    GROUP BY
+      field_group_type_value, field_person_target_id
+  ) b
+GROUP BY
+  field_group_type_value;
+  ")->fetchAllAssoc('type');
+}
+
+$encounters = [
+  [
+    'ANC (total)',
+    encounter_all_count('prenatal'),
+    encounter_unique_count('prenatal'),
+  ],
+  [
+    '   Health Center',
+    encounter_all_count('prenatal', 'hc'),
+    encounter_unique_count('prenatal', 'hc'),
+  ],
+  [
+    '   CHW',
+    encounter_all_count('prenatal') - encounter_all_count('prenatal', 'hc'),
+    encounter_unique_count('prenatal') - encounter_unique_count('prenatal', 'hc'),
+  ],
```

Comment:
Is it actually true @adamhstewart / @anvmn that an ANC / prenatal encounter is either health center or CHW, so everything which is not type 'nurse', is CHW or is it more complicated than  that?
![image](https://user-images.githubusercontent.com/114076/144437329-0cac303f-fb07-48f3-bf74-641cf756d0ea.png)
It seems so, just to double-check.

Reply:
Also, is it something that might change in the foreseeable future?

Reply:
@AronNovak yes, this is true, and AFAIK not supposed to change